src/test/java/SentinelTesting.java

```diff
@@ -158,4 +158,105 @@ public void test() {
 				                        """;
 		test(input, expectedOutput);
 	}
+
+	@Test
+	public void shadowingCheck() {
+		String input = """
+				public class SentinelTest {
+				    int val = 0;
+				    public void test() {
+				        String str = "Hello World";
+
+				        if (str == null) {
+				            val = -1;
+				        }
+
+				        if (val == 0) {
+				            System.out.println("Str is not null");
+				        }
+				    }
+
+				    public void shadow() {
+					int val = 0;
+				        if (val > 3) {
+						;
```

Comment:
This test case only tests shadowing in a different method than the one where we expect the refactoring to occur - that is, the "good" case. Can you also add a test for the "bad" case where the value is shadowed in a way that makes the refactoring unsafe, and show that your system doesn't apply the refactoring in that case? Such "negative" cases are often as important or more important to showing that a system is correct than the "good" test cases that show the system behaves as it should in the cases you've built it for, since many bugs are in the cases you haven't thought of :)